src/schema_impl/src/lib.rs

```diff
@@ -60,6 +60,8 @@ pub enum RegistrationError {
     InvalidSubscription(anyhow::Error),
     #[error("a subscription with the same id {0} already exists in the registry")]
     OverrideSubscription(EndpointId),
+    #[error("a schema update is not backwards compatible with the existing definition")]
+    IncompatibleSchemaEvolution(String, Vec<String>),
```

Comment:
Can you add this variant and DifferentServiceInstanceType to this match? https://github.com/restatedev/restate/blob/main/crates/meta/src/rest_api/error.rs#L72

With that, we make sure we return CONFLICT status code for those errors.